### PR TITLE
[1.x] Allow moneyphp version 4

### PR DESCRIPTION
**Pull request by mmachatschek**
This adds the same changes like https://github.com/laravel/cashier-stripe/pull/1280

### CODE REVIEW

**Review by driesvints — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
     ],
     "require": {
         "php": "^7.2|^8.0",
+        "ext-bcmath": "*",
```
Comment (GrahamCampbell) [inline composer.json:22]: Can this line be reverted? People who don't have this extension can still use moneyphp/money v3.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (driesvints) [reply composer.json:22]: No, this is required by moneyphp now and we also added this to Cashier Stripe. I don't expect this to have a huge impact as most PHP installs have this extension.

**Comment by driesvints** [comment]
@mmachatschek @GrahamCampbell is correct. `moneyphp/money` already requires bcmath so we don't need to require it through Cashier. I'll get the Stripe version fixed as well.

**Review by driesvints — APPROVED** [review]

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ jobs:
         uses: shivammathur/setup-php@v2
         with:
           php-version: ${{ matrix.php }}
-          extensions: dom, curl, libxml, mbstring, zip
+          extensions: dom, curl, libxml, mbstring, zip, bcmath
```
Comment (GrahamCampbell) [inline .github/workflows/tests.yml:34]: Would be nice to have these alphabetically sorted.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 2 above)
Comment (driesvints) [reply .github/workflows/tests.yml:34]: We don't do that anywhere. This is fine.

Comment (GrahamCampbell) [reply .github/workflows/tests.yml:34]: The original code has them sorted?

**Review by GrahamCampbell — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 2 above)
Comment (driesvints) [reply .github/workflows/tests.yml:34]: What original code?

Comment (driesvints) [reply .github/workflows/tests.yml:34]: It's not important. We don't do this anywhere.

**Pull request merged**